core/src/main/java/org/apache/iceberg/BaseTableScan.java

```diff
@@ -45,70 +38,46 @@
 /**
  * Base class for {@link TableScan} implementations.
  */
-abstract class BaseTableScan implements TableScan {
+abstract class BaseTableScan extends BaseScan<TableScan> implements TableScan {
   private static final Logger LOG = LoggerFactory.getLogger(BaseTableScan.class);
 
-  private final TableOperations ops;
-  private final Table table;
-  private final Schema schema;
-  private final TableScanContext context;
-
   protected BaseTableScan(TableOperations ops, Table table, Schema schema) {
     this(ops, table, schema, new TableScanContext());
   }
 
   protected BaseTableScan(TableOperations ops, Table table, Schema schema, TableScanContext context) {
-    this.ops = ops;
-    this.table = table;
-    this.schema = schema;
-    this.context = context;
-  }
-
-  protected TableOperations tableOps() {
-    return ops;
-  }
-
-  protected Schema tableSchema() {
-    return schema;
+    super(ops, table, schema, context);
   }
 
   protected Long snapshotId() {
-    return context.snapshotId();
+    return context().snapshotId();
   }
 
   protected boolean colStats() {
-    return context.returnColumnStats();
+    return context().returnColumnStats();
   }
 
   protected boolean shouldIgnoreResiduals() {
-    return context.ignoreResiduals();
+    return context().ignoreResiduals();
   }
 
   protected Collection<String> selectedColumns() {
```

Comment:
Is this still being used?

Reply:
doesn't looks like it is used based on code search. will remove it in the follow up